flake/modules/home-manager.nix

```diff
@@ -8,15 +12,17 @@ packages: lib: {
   inherit (lib.modules) mkIf mkAliasOptionModule;
   inherit (lib.lists) optional;
   inherit (lib.options) mkOption mkEnableOption literalExpression;
-  inherit (lib.types) attrsOf anything bool;
-  inherit (lib.nvim) neovimConfiguration;
-  inherit (lib.nvim.types) anythingConcatLists;
+  inherit (lib.types) anything bool submoduleWith;
 
   cfg = config.programs.nvf;
 
-  neovimConfigured = neovimConfiguration {
-    inherit pkgs;
-    modules = [cfg.settings];
+  nvfModule = submoduleWith {
+    description = "Nvf module";
+    class = "nvf";
+    specialArgs = {
+      inherit pkgs lib inputs;
+    };
+    modules = import ../../modules/modules.nix {inherit pkgs lib;};
```

Comment:
`pkgs` and `lib` are already in specialArgs, why do you need this import?


Wouldn't `modules = [../../modules/modules.nix];` do the trick here?

Reply:
modules/modules.nix is actually a list of modules rather than a single module... not sure why it's structured like that, should I change it?

Reply:
Something we've inherited from the original project, never got to changing it. Noticing now that it was inspired by home-manager.

I'll look into this again later today, you can keep it as is for now.